src/linker/Linker/Tracer.cs

```diff
@@ -37,6 +38,7 @@ public class Tracer
 
 		Stack<object> dependency_stack;
 		List<IDependencyRecorder> recorders;
+		List<IReflectionPatternRecorder> reflectionPatternRecorders;
```

Comment:
Why did you move the reflection pattern recorder into the Tracer? I don't see any reason for this?

Reply:
I was prototyping this with a graph-based IDependencyRecorder that reported paths to dangerous reflection sites. My recorder needed to know about the unanalyzed reflection patterns, so I set it up to work with a list of IReflectionPatternRecorders, at which point it seemed like it belonged in Tracer.cs - it really is another kind of tracing in my mind. I can put it in a separate PR if you prefer.

Reply:
I see - I would definitely prefer a separate PR - since I still don't see why needs to live on the tracer. I can see the point of view where it is another kind of traces - I'm not sure I like it though :-)